config/detekt/detekt.yml

```diff
@@ -365,7 +365,7 @@ naming:
     packagePattern: '[a-z]+(\.[a-z][A-Za-z0-9]*)*'
   TopLevelPropertyNaming:
     active: true
-    constantPattern: '[A-Z][A-Za-z0-9]*'
+    constantPattern: '[A-Z][_A-Za-z0-9]*'
```

Comment:
I'm not sure why underscores (`_`) weren't included in the constant pattern, but it seems incorrect to prohibit something like `MESHTASTIC_DOMAIN` in favor of `MESHTASTICDOMAIN`.

[Default Detekt configurations](https://detekt.dev/docs/rules/naming/#configuration-options-15) allow underscores, but I'm happy to change this back if it's preferable.

Reply:
Agree, feel free to adjust - it's a new add, and may need some tuning.

Reply:
Compose [introduced `PascalCase` convention](https://github.com/androidx/androidx/blob/androidx-main/compose/docs/compose-api-guidelines.md#singletons-constants-sealed-class-and-enum-class-values) but `CAPITALS_AND_UNDERSCORES` is also fine.

Reply:
fwiw, I actually think the compose documentation might be incorrect or out-of-date? The [Kotlin convention](https://kotlinlang.org/docs/coding-conventions.html#property-names) is shouty snake case (CAPITALS_AND_UNDERSCORES):
> Names of constants (properties marked with const, or top-level or object val properties with no custom get function that hold deeply immutable data) should use uppercase underscore-separated ([screaming snake case](https://en.wikipedia.org/wiki/Snake_case)) names.

Reply:
think of Kotlin guidelines as general rules and Compose guidelines as specific rules for Compose, which is a framework within Kotlin. So basically, what is mentioned in the Compose guidelines is what is different from Kotlin guidelines.

top-level constants used here are a common Compose pattern, so that's why Detekt wants `PascalCase` but since it doesn't really apply (ie. not Compose code) both should be allowed.